backend/django/core/views/api_annotate.py

```diff
@@ -555,7 +558,13 @@ def data_admin_table(request, project_pk):
         }
         data.append(temp)
 
-    return Response({"data": data})
+    messages = list(
```

Comment:
So I see we're sending the messages separately here from the data items they go with. Then on the frontend we have to search for the correct data message each time. This isn't necessarily wrong, but I'm wondering if we'll have runtime issues with more data. Is there a reason you decided to pull them this way instead of pulling the message object in the data loop above?